### PR TITLE
feat(theme-default): support full link for github repo

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

**Comment by meteorlxy** [comment]
The `repo` option should also be modified.

https://github.com/vuepress/vuepress-next/blob/5cad664bd7224a08e679dc06f61f17af6c790b97/packages/%40vuepress/theme-default/src/client/components/NavbarLinks.vue#L102-L107

File: packages/@vuepress/theme-default/src/client/utils/resolveEditLink.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const resolveEditLink = ({
   return pattern
     .replace(
       /:repo/,
-      repoType === 'GitHub' ? `https://github.com/${docsRepo}` : docsRepo
+      docsRepo.startsWith('http') ? docsRepo : `https://github.com/${docsRepo}`
```
Comment (meteorlxy) [inline packages/@vuepress/theme-default/src/client/utils/resolveEditLink.ts:40]: Should make use of the `isLinkHttp` util

**Review by meteorlxy — CHANGES_REQUESTED** [review]

File: packages/@vuepress/theme-default/src/client/utils/resolveEditLink.ts (hunk 1 above)
Comment (Mister-Hope) [reply packages/@vuepress/theme-default/src/client/utils/resolveEditLink.ts:40]: Gotit

**Comment by meteorlxy** [comment]
The corresponding docs should be updated, too.

**Comment by meteorlxy** [comment]
Ummmm, what I mean is this part:

https://github.com/vuepress/vuepress-next/blob/a38f2a246cffeab3ba78a9f728303c430b9fae5d/docs/reference/default-theme/config.md#L149

**Comment by meteorlxy** [comment]
I'll update it

**Review by meteorlxy — APPROVED** [review]

**Pull request merged**